server/src/main/java/io/spine/server/commandbus/CommandFlightRecorder.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.commandbus;
+
+import io.spine.base.EventMessage;
+import io.spine.core.CommandContext;
+import io.spine.server.type.CommandEnvelope;
+import io.spine.system.server.SystemWriteSide;
+import io.spine.system.server.WriteSideFunction;
+import io.spine.system.server.event.CommandDispatched;
+import io.spine.system.server.event.CommandScheduled;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Records a command log by posting the corresponding events into the
+ * {@link io.spine.system.server.SystemContext system context}.
+ */
+final class CommandFlightRecorder implements CommandFlowWatcher {
```

Comment:
As an afterthought, since this class is package-private in `commandbus`, we could just call it `FlightRecorder` without the `Command` prefix.